packages/next/src/build/webpack/plugins/next-types-plugin.ts

```diff
@@ -118,51 +122,64 @@ async function collectNamedSlots(layoutPath: string) {
   return slots
 }
 
-const nodeRouteTypes: string[] = []
-const edgeRouteTypes: string[] = []
+let edgeRouteTypes = new Set<string>()
+let nodeRouteTypes = new Set<string>()
```

Comment:
What's the benefit of using Set here? To handle conflict routes?

Reply:
@shuding I want to make sure that nodeRouteTypes doesn't cause routeTypes to have duplicated pages (pages that also appear in edgeRouteTypes), and I think Set.has is better than Array.indexOf and stuffs like that :)

Reply:
`nodeRouteTypes` and `edgeRouteTypes` won't have any shared routes (because one route must be either the Node runtime or the Edge runtime), and I think they can be just arrays?

Reply:
@shuding that is the expected behaviour, yes, but it seems that both edgeRouteTypes and nodeRouteTypes just have all the routes in them - perhaps we should check something else, rather than `this.isEdgeServer`.

Reply:
Then that sounds like another bug and we will tackle separately! Thanks